include/zephyr/net/wifi_mgmt.h

```diff
@@ -1219,14 +1219,14 @@ struct wifi_wps_config_params {
 
 /** Wi-Fi AP status
  */
-enum wifi_hostapd_iface_state {
-	WIFI_HAPD_IFACE_UNINITIALIZED,
-	WIFI_HAPD_IFACE_DISABLED,
-	WIFI_HAPD_IFACE_COUNTRY_UPDATE,
-	WIFI_HAPD_IFACE_ACS,
-	WIFI_HAPD_IFACE_HT_SCAN,
-	WIFI_HAPD_IFACE_DFS,
-	WIFI_HAPD_IFACE_ENABLED
+enum wifi_sap_iface_state {
```

Comment:
This is good. We should abstract out implementation (NM/offload) as much as possible. Even these states directly map to hostapd, but may be we can leave it as is, but can revisit to see if we need abstract these out as well.